jpf-symbc/src/examples/TestMathMaxMin.java

```diff
@@ -0,0 +1,106 @@
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This class tests the behaviour of {@code Math.max()} and {@code Math.min()} methods.
+ * <p>
+ *     Edge case handling of {@code NaN} and {@code 0.0/-0.0}
+ * </p>
+ */
+
+public class TestMathMaxMin {
+
+    /**
+     * This method tests {@code Math.max()} for {@code double} type values.
+     */
+
+    public static void testDoubleMax() {
+
+        /* This will pass as exact bit representations of 0.0 and -0.0 are not equal.
+           But earlier If we kept -0.0 as the first argument it was returning -0.0, which is wrong. */
+        assert Double.doubleToRawLongBits(Math.max(-0.0, 0.0)) == Double.doubleToRawLongBits(0.0) : "Test case Failed as Expected";
+
+        // This will fail
+        assert Double.doubleToRawLongBits(Math.max(0.0, -0.0)) == Double.doubleToRawLongBits(-0.0) : "Test case Failed";
+
+        /* This will fail as expected which is the right behaviour, earlier Math.max() returned 5.0
+           Similar to the case above. */
+        assert Double.doubleToRawLongBits(Math.max(Double.NaN, 5.0)) == Double.doubleToRawLongBits(5.0) : "Test case Failed as Expected";
+
+        // Eventually this will pass now
+        assert Double.doubleToRawLongBits(Math.max(5.0, Double.NaN)) == Double.doubleToRawLongBits(Double.NaN) : "Test case Failed";
+
+        // Additional Test Cases. Similar to the cases above.
+        System.out.println(Math.max(Double.NaN, Double.POSITIVE_INFINITY)); // earlier this returned +Infinity, but should be NaN
+        System.out.println(Math.max(Double.NaN, Double.NEGATIVE_INFINITY)); // earlier this returned -Infinity, but should be NaN
+
+    }
+
+
+    /**
+     * This method tests {@code Math.max()} for {@code float} type values.
+     */
+
+    public static void testFloatMax() {
+
+        // Tests similar to the cases above.
+        System.out.println(Math.max(0.0f, -0.0f)); // 0.0
+        System.out.println(Math.max(-0.0f, 0.0f)); // 0.0
+        System.out.println(Math.max(Float.NaN, 4.5)); // NaN
+        System.out.println(Math.max(4.5, Float.NaN)); // NaN
+    }
+
+
+    /**
+     * This method tests {@code Math.min()} for {@code double} type values.
+     */
+
+    public static void testDoubleMin() {
+
+        // Earlier this was returning 0.0, -0.0, 4.5, NaN and 9.0
+        System.out.println(Math.min(0.0, -0.0)); // 0.0
+        System.out.println(Math.min(-0.0, 0.0)); // 0.0
+        System.out.println(Math.min(Double.NaN, 4.5)); // NaN
+        System.out.println(Math.min(4.5, Double.NaN)); // NaN
+        System.out.println(Math.min(5.0, 9.0)); // 5.0
```

Comment:
Can you please make these a set of assert statements as you've done with testDouble Max?